_source/_posts/2022-06-01-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,604 @@
+---
+layout: blog_post
+title: "Cloud Native Java Microservices with JHipster and Istio"
+author: deepu-sasidharan
+by: advocate
+communities: [devops, kubernetes, java]
+description: "Build a cloud-native Java microservice stack on Google Kubernetes Engine using Istio and JHipster."
+tags: [kubernetes, jhipster, spring-boot, spring, java, istio, gcp, react, service-mesh]
+tweets:
+  - "Build a #Java microservice stack on Google Cloud using #JHipster, #Istio, and #Kubernetes"
+  - "Deploy a cloud-native #Java microservice stack to GKE with #Istio and #JHipster"
+image:
+type: awareness
+---
+
+Microservices are not everyone's cup of tea, and they shouldn't be. Not every problem can be or should be solved by microservices. Sometimes building a simple monolith is a far better option. Microservices are solutions for use cases where scale and scalability are important. A few years ago, microservices were all the rage, made popular, especially by companies like Netflix, Spotify, Google, etc. While the hype has died down a bit, the genuine use cases still exist, and with the advances in cloud computing technologies, building microservices as cloud-native services is the way to go due to many of its benefits.
+
+Today we will look at building a cloud-native Java microservice stack that utilizes a service mesh to provide most of the distributed system needs and will deploy it to the cloud using Kubernetes.
+
+So here is what we will be doing today:
+
+- Build a Java microservice stack using JHipster, Spring Boot, and Spring Cloud
+- Create a Google Kubernetes Engine (GKE) cluster
+- Deploy Istio service mesh to the cluster
+- Setup monitoring and observability
+- Deploy and monitor the microservices to the cluster
+
+Let's get started!
+
+{% include toc.md %}
+
+If you prefer to follow along by watching a video, here's the video from the [OktaDev YouTube channel](https://youtu.be/zGpnIhRgMaM).
+
+{% youtube zGpnIhRgMaM %}
+
+**Pre-requisites**
```

Comment:
```suggestion
**Prerequisites**
```

Reply:
FWIW, I usually put the prerequisites right before the TOC. For example, see [this post](https://developer.okta.com/blog/2022/06/02/java-rest-api-helidon).